storage/invoke_test.go

```diff
@@ -203,6 +242,10 @@ func TestInvoke(t *testing.T) {
 			if !test.expectFinalErr {
 				wantAttempts = 1
 			}
+			if test.retry != nil && test.retry.maxAttempts != 0 && test.retry.policy != RetryNever {
+				wantAttempts = 1 + test.retry.maxAttempts
```

Comment:
This seems off-by-1. After thinking about this more I think with the name `MaxAttempts` we should have the semantics be the number of total calls including all retries. So if the user says `WithMaxAttempts(3)` then we do up to 3 total calls (2 retries). Does this make sense to you?

If so let's 1. clarify the docs 2. Set L73 in invoke to be `>= rather than `>` 3. remove the `1 +` here.

Reply:
I think with 3 retries configured `(WithMaxAttempts(3))`, the function will be called a maximum of 4 times. The first call is the initial attempt, and the subsequent calls are retries triggered by failures.